src/WebJobs.Script/Workers/Rpc/Configuration/WorkerConfigurationResolverOptionsSetup.cs

```diff
@@ -62,41 +100,192 @@ private string GetWorkersRootDirPath(IConfiguration configuration)
+
+                    configuration = new ConfigurationBuilder()
+                        .AddConfiguration(_configuration)
+                        .AddConfiguration(latestConfiguration)
+                        .Build();
+                }
+            }
+
+            return configuration;
+        }
+
+        /// <summary>
+        /// Logs a message if the required configuration section is found.
+        /// </summary>
+        private void LogWorkersDirSectionPresence(IConfiguration configuration, string configurationSource)
+        {
+            string configSectionToCheck = ConfigurationPath.Combine(RpcWorkerConstants.LanguageWorkersSectionName, WorkerConstants.WorkersDirectorySectionName);
+            var section = configuration.GetSection(configSectionToCheck);
+
+            if (!string.IsNullOrEmpty(section.Value))
+            {
+                _logger.LogDebug("Found configuration section '{requiredSection}' in {configurationSource}.", configSectionToCheck, configurationSource);
+            }
+        }
+
+        /// <summary>
+        /// Retrieves the list of worker probing paths from configuration or uses the default path for Windows environment.
+        /// </summary>
+        internal List<string> GetWorkerProbingPaths(IConfiguration configuration)
+        {
+            // If Configuration section is set, read probing paths from configuration.
+            var probingPathsSection = configuration.GetSection(ConfigurationPath.Combine(RpcWorkerConstants.LanguageWorkersSectionName, RpcWorkerConstants.WorkerProbingPathsSectionName));
+            var probingPaths = probingPathsSection.Get<List<string>>();
+            _logger.LogDebug("Worker probing paths specified via configuration: {probingPaths}.", probingPaths);
+
+            probingPaths = probingPaths ?? [];
+
+            if (probingPaths.Count == 0)
+            {
+                if (_environment.IsHostedWindowsEnvironment())
+                {
+                    // Default worker probing path for Windows
+                    string windowsSiteExtensionsPath = GetWindowsSiteExtensionsPath();
+
+                    if (!string.IsNullOrWhiteSpace(windowsSiteExtensionsPath))
+                    {
+                        // Example probing path for Windows: "c:\\home\\SiteExtensions\\functionsworkers"
+                        var windowsWorkerProbingPath = Path.Combine(windowsSiteExtensionsPath, RpcWorkerConstants.FunctionsWorkersDirectoryName);
+                        probingPaths.Add(windowsWorkerProbingPath);
+                        _logger.LogDebug("Default worker probing path for Windows: {windowsWorkerProbingPath}.", windowsWorkerProbingPath);
+                    }
+                }
+            }
+
+            return probingPaths;
+        }
+
+        /// <summary>
+        /// Returns the default site extensions path for Windows environment.
+        /// </summary>
+        private string GetWindowsSiteExtensionsPath()
```

Comment:
```suggestion
        private static string GetWindowsSiteExtensionsPath()
```

Marking methods as static can provide a small perf benefit at runtime. Please review the other methods in this PR to see if any of them could also be marked as static.

Reply:
https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1822#rule-description